### ISSUE TITLE
Add all versioned package to the pin file

### ISSUE THREAD

**Issue by bsipocz** [astropy/ci-helpers#275]
E.g. even though PYTHON_VERSION=3.4 is set, using conda 4.4.10 upgrades it to python 3.6, so probably we need to make sure to pin each and every package version that was specifically set either as dependency or by individual env variable.

e.g.:
https://travis-ci.org/astropy/ci-helpers/jobs/343116908

**Comment by kain88-de** [comment]
Since when is this a problem? I would like to check the build-logs of MDAnalysis. We are using conda-forge as our main channel and I want to know if we are affected by that too.

**Comment by bsipocz** [comment]
I don't think the python version issue was ever a problem in production, the ci-helpers tests are checking for it and never failed before. Actually that is how I realized that with the new conda it may be a problem.

However in some corner cases I saw that packages are downgraded, most often when an old version number is pinned for a dependency and it's not compatible any more with the latest and greatest. This is the reason we pin the latest version numbers, e.g. for numpy and astropy, rather than just let conda pick a version it likes.
